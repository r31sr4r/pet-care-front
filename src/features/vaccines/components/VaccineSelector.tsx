import { FormControl, InputLabel, MenuItem } from '@mui/material';
import Select, { SelectChangeEvent } from '@mui/material/Select';
import { useGetVaccinesByBreedTypeQuery } from '../vaccinesSlice';

type Props = {
	vaccineName: string;
	breedType: string;
	handleVaccineChange: (e: SelectChangeEvent) => void;
};

export function VaccineSelector({
	vaccineName,
	breedType,
	handleVaccineChange,
}: Props) {
	console.log('breedType', breedType);
	const { data, isFetching, error } = useGetVaccinesByBreedTypeQuery({
		breed_type: breedType,
	});
	return (
		<FormControl fullWidth>
			<InputLabel id="demo-simple-select-label">Vacina</InputLabel>
			<Select
				name="vaccine"
				labelId="demo-simple-select-label"
				id="demo-simple-select"
				value={ isFetching ? '' : vaccineName }
				label="Vacina"
				onChange={handleVaccineChange}
				key={vaccineName}
			>
				{data?.data
					? data.data.map((vaccine: any) => (
							<MenuItem value={vaccine.name} key={vaccine.id}>
								{vaccine.name}
							</MenuItem>
					  ))
					: []}
			</Select>
		</FormControl>
	);
}
