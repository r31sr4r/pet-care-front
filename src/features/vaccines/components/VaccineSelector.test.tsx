import { renderWithProviders } from '../../../utils/test/test-utils';
import { VaccineSelector } from './VaccineSelector';

const breedType = 'DOG';

const Props = {
    breedType: breedType,
    vaccineName: '',
    handleVaccineChange: () => {},
};

describe('VaccineSelector', () => {

    it('should render successfully', () => {
        const { asFragment } = renderWithProviders(<VaccineSelector {...Props} />);
        expect(asFragment()).toMatchSnapshot();        
    });
});