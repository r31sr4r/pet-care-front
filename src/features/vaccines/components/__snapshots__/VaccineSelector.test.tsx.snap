// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaccineSelector should render successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      id="demo-simple-select-label"
    >
      Vacina
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-label demo-simple-select"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="demo-simple-select"
        role="button"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        name="vaccine"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            Vacina
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
